src/neo4j_graphrag/experimental/components/entity_relation_extractor.py

```diff
@@ -100,28 +102,19 @@ def balance_curly_braces(json_string: str) -> str:
     return "".join(fixed_json)
 
 
-def fix_invalid_json(invalid_json_string: str) -> str:
-    # Fix missing quotes around field names
-    invalid_json_string = re.sub(
-        r"([{,]\s*)(\w+)(\s*:)", r'\1"\2"\3', invalid_json_string
-    )
-
-    # Fix missing quotes around string values, correctly ignoring null, true, false, and numeric values
-    invalid_json_string = re.sub(
-        r"(?<=:\s)(?!(null|true|false|\d+\.?\d*))([a-zA-Z_][a-zA-Z0-9_]*)\s*(?=[,}])",
-        r'"\2"',
-        invalid_json_string,
-    )
-
-    # Correct the specific issue: remove trailing commas within arrays or objects before closing braces or brackets
-    invalid_json_string = re.sub(r",\s*(?=[}\]])", "", invalid_json_string)
+def fix_invalid_json(raw_json: str) -> str:
+    repaired_json = json_repair.repair_json(raw_json)
 
-    # Normalize excessive curly braces
-    invalid_json_string = re.sub(r"{{+", "{", invalid_json_string)
-    invalid_json_string = re.sub(r"}}+", "}", invalid_json_string)
+    if isinstance(repaired_json, str):
```

Comment:
In which cases it's not a string?

Reply:
Looks like it can be: [JSONReturnType](https://github.com/mangiucugna/json_repair/blob/main/src/json_repair/json_parser.py#L6)

Do you think this check is appropriate?

Reply:
I see. However, the `repair_json` function returns a `json.dumps`, so it will always be a string in our case (https://github.com/mangiucugna/json_repair/blob/main/src/json_repair/json_repair.py#L72)

If mypy is complaining, maybe we can use `cast` here as well?

Reply:
Hmm in this case wouldn't it better to have a runtime check using `isinstance`? In case this implementation changes or `repair_json` is incorrectly used

Reply:
This is not a part that we let to developers, it's our code. So for now we know how it behaves and if we change it later on, it will be the time to add checks. Same for a potential change in a later version of the library. Basically my view is: let's not add complexity to cover future potential issues, that we are not even sure will ever happen.

Reply:
https://github.com/neo4j/neo4j-graphrag-python/pull/226/commits/639cae231ea040dd92053a5637c403208e0a0e04